addons/binding/org.openhab.binding.hyperion/src/main/java/org/openhab/binding/hyperion/handler/HyperionNgHandler.java

```diff
@@ -0,0 +1,494 @@
+    private static final String COMPONENTS_ALL = "ALL";
+
+    private JsonTcpConnection connection;
+    private ScheduledFuture<?> refreshFuture;
+    private ScheduledFuture<?> connectFuture;
+    private Gson gson = new Gson();
+
+    private static final ServerInfoCommand SERVER_INFO_COMMAND = new ServerInfoCommand();
+
+    private String address;
+    private int port;
+    private int refreshInterval;
+    private int priority;
+    private String origin;
+    private HyperionStateDescriptionProvider stateDescriptionProvider;
+
+    private Runnable refreshJob = new Runnable() {
+        @Override
+        public void run() {
+            try {
+                NgResponse response = sendCommand(SERVER_INFO_COMMAND);
+                if (response.isSuccess()) {
+                    handleServerInfoResponse(response);
+                }
+                updateOnlineStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE);
+            } catch (IOException e) {
+                updateOnlineStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            } catch (JsonParseException e) {
+                logger.warn("{}", e.getMessage(), e);
+            } catch (CommandUnsuccessfulException e) {
+                logger.warn("Server rejected the command: {}", e.getMessage());
+            }
+        }
+    };
+
+    private Runnable connectionJob = new Runnable() {
+        @Override
+        public void run() {
+            try {
+                if (!connection.isConnected()) {
+                    connection.connect();
+                    updateOnlineStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE);
+                }
+            } catch (IOException e) {
+                updateOnlineStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            }
+        }
+    };
+
+    public HyperionNgHandler(Thing thing, HyperionStateDescriptionProvider stateDescriptionProvider) {
+        super(thing);
+        this.stateDescriptionProvider = stateDescriptionProvider;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Hyperion.ng thing handler.");
+        try {
+            Configuration config = thing.getConfiguration();
+            address = (String) config.get(PROP_HOST);
+            port = ((BigDecimal) config.get(PROP_PORT)).intValue();
+            refreshInterval = ((BigDecimal) config.get(PROP_POLL_FREQUENCY)).intValue();
+            priority = ((BigDecimal) config.get(PROP_PRIORITY)).intValue();
+            origin = (String) config.get(PROP_ORIGIN);
+
+            connection = new JsonTcpConnection(address, port);
+            connectFuture = scheduler.scheduleWithFixedDelay(connectionJob, 0, refreshInterval, TimeUnit.SECONDS);
+            refreshFuture = scheduler.scheduleWithFixedDelay(refreshJob, 0, refreshInterval, TimeUnit.SECONDS);
+        } catch (UnknownHostException e) {
+            logger.warn("Could not resolve host: {}", e.getMessage());
```

Comment:
all other occurences as well - please reduce to debug. I won't continue to comment on that from here

Reply:
done